ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -425,6 +426,8 @@ private PlanNode buildSourceNode() {
 
     if (sources.size() == 1) {
       throw new IllegalStateException("Expected more than one source. Got " + sources.size());
+    } else if (sources.size() != 2) {
+      throw new KsqlException("ksqlDB does not support multi-way joins.");
```

Comment:
Is this part of a followup?

Reply:
yup! I wanted to split it up into smaller PRs